src/azul/chalice.py

```diff
@@ -201,6 +201,19 @@ def _security_headers_middleware(self, event, get_response):
         response.headers['Strict-Transport-Security'] = f'max-age={seconds}; includeSubDomains'
         response.headers['X-Content-Type-Options'] = 'nosniff'
         response.headers['X-Frame-Options'] = 'DENY'
+        path = event.context['path']
+        if path == '/':
+            cache_control = 'public, max-age=0, must-revalidate'
+        elif path.startswith('/static/'):
+            cache_control = 'public, max-age=86400'
+        elif path == '/openapi':
+            cache_control = 'public, max-age=500'
+        elif path == '/oauth2_redirect':
+            # Set explicitly in case the default ever changes
+            cache_control = 'no-store'
+        else:
+            cache_control = 'no-store'
+        response.headers['Cache-Control'] = cache_control
```

Comment:
It's OK to set a default here but the path-specific cases should be set in each route handler.